File: HTTP/Go/transmission.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"time"
)

type Recipient struct {
	Address        string            `json:"address"`
	Name           string            `json:"name"`
	AdditionalData map[string]string `json:"additional_data"`
	MergeData      map[string]string `json:"merge_data"`
}

type Content struct {
	Subject string `json:"subject"`
	HTML    string `json:"html"`
	Text    string `json:"text"`
	From    struct {
		Address string `json:"address"`
		Name    string `json:"name"`
	} `json:"from"`
}

type Payload struct {
	CampaignName string      `json:"campaign_name"`
	Recipients   []Recipient `json:"recipients"`
	Content      Content     `json:"content"`
}

func main() {
	accessToken := "1000.****************************"

	// Recipients
	recipients := []Recipient{
		{
			Address: "lucy@example.campaigns.zoho.com",
			Name:    "Aaron Fletcher",
			AdditionalData: map[string]string{
				"phone":   "+919876543210",
				"country": "IN",
			},
			MergeData: map[string]string{
				"first_name": "Aaron",
			},
		},
	}

	// Content
	content := Content{
		Subject: "My first mail using Zoho Campaigns Email API HTTP",
		HTML:    "<html><body>Welcome $[first_name|Customer]$!<br>Summer Hot Savings, You Don't Want to Miss</body></html>",
		Text:    "Welcome $[first_name|Customer]$! Summer Hot Savings, You Don’t Want to Miss",
	}
	content.From.Address = "aron@marketing.campaigns.zoho.com"
	content.From.Name = "Aron Fletcher"

	// Payload
	payload := Payload{
		CampaignName: "hello_customer",
		Recipients:   recipients,
		Content:      content,
	}

	// Encode payload to JSON
	payloadBytes, err := json.Marshal(payload)
	if err != nil {
		fmt.Printf("Error marshalling payload: %v\n", err)
		return
	}

	// Set up HTTP request
	url := "https://campaigns.zoho.com/emailapi/v1/transmission"
	req, err := http.NewRequest("POST", url, bytes.NewBuffer(payloadBytes))
	if err != nil {
		fmt.Printf("Error creating request: %v\n", err)
		return
	}

	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("Accept", "application/json")
	req.Header.Set("Authorization", "Zoho-oauthtoken "+accessToken)

	client := &http.Client{
		Timeout: 60 * time.Second,
	}

	// Send HTTP request
	resp, err := client.Do(req)
	if err != nil {
		fmt.Printf("Error sending request: %v\n", err)
		return
	}
	defer resp.Body.Close()

	// Read response
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("Error reading response body: %v\n", err)
		return
	}

	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
		fmt.Printf("Response: %s\n", body)
	} else {
		fmt.Printf("Error: %s\n", body)
	}
}

File: HTTP/Node.js/transmission.js
const axios = require('axios');

async function main() {
    const accessToken = "1000.****************************";

    // Recipients
    const recipientData = [{
        address: "lucy@example.campaigns.zoho.com",
        name: "Aaron Fletcher",
        additional_data: {
            phone: "+919876543210",
            country: "IN"
        },
        merge_data: {
            first_name: "Aaron"
        }
    }];

    // Content
    const content = {
        subject: "My first mail using Zoho Campaigns Email API HTTP",
        html: "<html><body>Welcome $[first_name|Customer]$!<br>Summer Hot Savings, You Don't Want to Miss</body></html>",
        text: "Welcome $[first_name|Customer]$! Summer Hot Savings, You Don’t Want to Miss",
        from: {
            address: "aron@marketing.campaigns.zoho.com",
            name: "Aron Fletcher"
        }
    };

    // Payload
    const payload = {
        campaign_name: "hello_customer",
        recipients: recipientData,
        content: content
    };

    try {
        const url = "https://campaigns.zoho.com/emailapi/v1/transmission";
        const headers = {
            "Content-Type": "application/json",
            "Accept": "application/json",
            "Authorization": `Zoho-oauthtoken ${accessToken}`
        };

        const response = await axios.post(url, payload, { headers, timeout: 60000 });

        if (response.status === 200) {
            console.log("Response:", JSON.stringify(response.data, null, 2));
        } else {
            console.log(`Error: ${response.status}`);
            console.log(response.data);
        }
    } catch (error) {
        console.error(`Exception occurred: ${error}`);
        if (error.response) {
            console.error(`Status: ${error.response.status}`);
            console.error(error.response.data);
        }
    }
}

main();
